cmd/cosign/cli/verify/verify_blob.go

```diff
@@ -94,6 +93,10 @@ func (c *VerifyBlobCmd) Exec(ctx context.Context, blobRef string) error {
 		return &options.PubKeyParseError{}
 	}
 
+	if c.CertRef != "" && (c.CertIdentity == "" || c.CertOIDCIssuer == "") {
```

Comment:
I don't *love* the duplication here. Could we pull this out into a common "VerificationIdentity" struct that goes into the `VerifyBlobCmd` (and other `*BlobCmd` structs) and stick the check there?

Reply:
I moved the check into CheckVerifyOptions. WDYT? I'm not convinced it's the best solution, but it _is_ already referenced by all the Check*Cmd structs.

Reply:
I like this overall, I think. I'm finding it more readable.

I was imagining sticking the `KeyRef` inside `CertVerifyOptions` as well so we wouldn't have to duplicate the `if KeyRef != ""` check. Then, the `Identities()` would return basically a sum-type of `cosign.Identity | PublicKey` you could `switch` over that. WDYT?

Part of what you're running up against is that this whole thing needs a big refactor (not your responsibility), gross. Thanks for your patience :)

Reply:
I'm not sure I like that solution since it's pulling (a tiny) part of key parsing into a struct that's about certificate verification.

I filed https://github.com/sigstore/cosign/issues/2462 with my suggestions for how to go about the larger refactor. How about we deal with the duplication in this PR and then I address the larger refactor in a follow-on change?